src/main/java/org/broadinstitute/dsm/jobs/UPSTrackingJob.java

```diff
@@ -21,60 +22,230 @@
+            }
+            else {
+                logError(trackingId, response.getErrors());
+            }
+        }
+        catch (IOException e) {
+            throw new RuntimeException("couldn't get response from ups tracking ", e);
+        }
+    }
+
+    private static void logError(String trackingId, UPSError[] errors) {
+        String errorString = "";
+        for (UPSError error : errors) {
+            errorString += "Got Error: " + error.getCode() + " " + error.getMessage() + " For Tracking Number " + trackingId;
+        }
+        logger.error(errorString);
+    }
+
+    private static void updateStatus(String trackingId, String oldType, UPSTrackingResponse response, boolean isReturn, DdpKit kit) {
+        if (response.getTrackResponse() != null) {
+            UPSShipment[] shipment = response.getTrackResponse().getShipment();
+
+            if (shipment != null && shipment.length > 0) {
+                UPSPackage[] upsPackages = shipment[0].getUpsPackageArray();
+                if (upsPackages != null && upsPackages.length > 0) {
+                    UPSPackage upsPackage = upsPackages[0];
+                    UPSActivity activity = upsPackage.getActivity()[0];
+                    if (activity != null) {
+                        UPSStatus status = activity.getStatus();
+                        if (status != null) {
+                            String statusType = status.getType();
+                            String statusDescription = status.getDescription();
+                            String date = activity.getDate() + " " + activity.getTime();
+                            if (!isReturn) {
+                                updateTrackingInfo(statusType, oldType, statusDescription, trackingId, date, SQL_UPDATE_UPS_TRACKING_STATUS, isReturn, kit);
+                            }
+                            else {
+                                updateTrackingInfo(statusType, oldType, statusDescription, trackingId, date, SQL_UPDATE_UPS_RETURN_STATUS, isReturn, kit);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+
+    private static void updateTrackingInfo(String statusType, String oldType, String statusDescription, String trackingId, String date, String query, boolean isReturn, DdpKit kit) {
+        String upsUpdate = statusType + " " + statusDescription;
+        SimpleResult result = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(query)) {
+                stmt.setString(1, upsUpdate);
+                stmt.setString(2, date);
+                stmt.setString(3, trackingId);
+                int r = stmt.executeUpdate();
+                if (r != 2) {//number of subkits
+                    throw new RuntimeException("Update query for UPS tracking updated " + r + " rows!");
```

Comment:
Maybe add some more context to this error?  `trackingId` or whatever is a handy value from `kit`.